test_scripts/Polices/App_Permissions/ATF_HP_Order_Of_Request_Response_Notification_On_Registering_App.lua

```diff
@@ -0,0 +1,143 @@
+config.defaultProtocolVersion = 2
+
+--[[ Required Shared libraries ]]
+local commonFunctions = require ('user_modules/shared_testcases/commonFunctions')
+local commonSteps = require ('user_modules/shared_testcases/commonSteps')
+
+--[[ General Settings for configuration ]]
+Test = require('connecttest')
+require('cardinalities')
+local mobile_session = require('mobile_session')
+
+--[[ Local Functions ]]
+local function Precondition()
+
+  commonFunctions:userPrint(34, "-- Precondition --")
+
+  function Test:StopSDL()
+    StopSDL()
+  end
+
+  --ToDo: shall be removed when issue: "SDL doesn't stop at execution ATF function StopSDL()" is fixed
+  function Test:SDLForceStop()
+    commonFunctions:SDLForceStop()
+  end
+
+  function Test:DeleteLogsAndPolicyTable()
+    commonSteps:DeleteLogsFiles()
+    commonSteps:DeletePolicyTable()
+  end
+
+  function Test:StartSDL_FirstLifeCycle()
+    StartSDL(config.pathToSDL, config.ExitOnCrash)
+  end
+
+  function Test:InitHMI_FirstLifeCycle()
+    self:initHMI()
+  end
+
+  function Test:InitHMI_onReady_FirstLifeCycle()
+    self:initHMI_onReady()
+  end
+
+  function Test:ConnectMobile_FirstLifeCycle()
+    self:connectMobile()
+  end
+
+  function Test:StartSession()
+    self.mobileSession = mobile_session.MobileSession(self, self.mobileConnection)
+    self.mobileSession:StartService(7)
+  end
+
+end
+
+--[[ Preconditions ]]
+Precondition()
+
+--[[ Test ]]
+function Test:Register_App_And_Check_Order_Of_Request_Response_Notiofications()
+  commonFunctions:userPrint(34, "-- Test --")
+  local CorIdRAI = self.mobileSession:SendRPC("RegisterAppInterface",
+    {
+      syncMsgVersion =
+      {
+        majorVersion = 3,
+        minorVersion = 0
+      },
+      appName = "SPT",
+      isMediaApplication = true,
+      languageDesired = "EN-US",
+      hmiDisplayLanguageDesired = "EN-US",
+      appID = "1234567",
+      deviceInfo =
+      {
+        os = "Android",
+        carrier = "Megafon",
+        firmwareRev = "Name: Linux, Version: 3.4.0-perf",
+        osVersion = "4.4.2",
+        maxNumberRFCOMMPorts = 1
+      }
+    })
+
+  EXPECT_HMINOTIFICATION("BasicCommunication.OnAppRegistered",
+    {
+      application =
+      {
+        appName = "SPT",
+        policyAppID = "1234567",
+        isMediaApplication = true,
+        hmiDisplayLanguageDesired = "EN-US",
+        deviceInfo =
+        {
+          name = "127.0.0.1",
+          id = config.deviceMAC,
+          transportType = "WIFI",
+          isSDLAllowed = false
+        }
+      }
+    })
+  :Do(function(_,data)
+      self.applications["SPT"] = data.params.application.appID
+    end)
+
+  EXPECT_RESPONSE(CorIdRAI, { success = true, resultCode = "SUCCESS"})
+  EXPECT_NOTIFICATION("OnHMIStatus", {hmiLevel = "NONE", audioStreamingState = "NOT_AUDIBLE", systemContext = "MAIN"})
+  EXPECT_NOTIFICATION("OnPermissionsChange", {})
```

Comment:
@OVikhrov, just suggestion: maybe there is the point to specify in **EXPECT_NOTIFICATION("OnPermissionsChange", {})** params from "pre_DataConsent" section.